Mathlib/Algebra/Group/Equiv/Basic.lean

```diff
@@ -80,10 +81,61 @@ def arrowCongr {M N P Q : Type*} [Mul P] [Mul Q] (f : M ≃ N) (g : P ≃* Q) :
   right_inv k := by ext; simp
   map_mul' h k := by ext; simp
 
+section monoidHomCongrEquiv
+variable [MulOneClass M] [MulOneClass M₁] [MulOneClass M₂] [Monoid N] [Monoid N₁] [Monoid N₂]
+
+/-- The equivalence `(M₁ →* N) ≃ (M₂ →* N)` obtained by postcomposition with
+a multiplicative equivalence `e : M₁ ≃* M₂`. -/
+@[to_additive (attr := simps)
+"The equivalence `(M₁ →+ N) ≃ (M₂ →+ N)` obtained by postcomposition with
+an additive equivalence `e : M₁ ≃+ M₂`."]
+def monoidHomCongrLeftEquiv (e : M₁ ≃* M₂) : (M₁ →* N) ≃ (M₂ →* N) where
+  toFun f := f.comp e.symm.toMonoidHom
+  invFun f := f.comp e.toMonoidHom
+  left_inv f := by ext; simp
+  right_inv f := by ext; simp
+
+/-- The equivalence `(M →* N₁) ≃ (M →* N₂)` obtained by postcomposition with
+a multiplicative equivalence `e : N₁ ≃* N₂`. -/
+@[to_additive (attr := simps)
+"The equivalence `(M →+ N₁) ≃ (M →+ N₂)` obtained by postcomposition with
+an additive equivalence `e : N₁ ≃+ N₂`."]
+def monoidHomCongrRightEquiv (e : N₁ ≃* N₂) : (M →* N₁) ≃ (M →* N₂) where
+  toFun := e.toMonoidHom.comp
+  invFun := e.symm.toMonoidHom.comp
+  left_inv f := by ext; simp
+  right_inv f := by ext; simp
+
+end monoidHomCongrEquiv
+
+section monoidHomCongr
+variable [MulOneClass M] [MulOneClass M₁] [MulOneClass M₂] [CommMonoid N] [CommMonoid N₁]
+  [CommMonoid N₂]
+
+/-- The isomorphism `(M₁ →* N) ≃* (M₂ →* N)` obtained by postcomposition with
+a multiplicative equivalence `e : M₁ ≃* M₂`. -/
+@[to_additive (attr := simps! toEquiv apply)
+"The isomorphism `(M₁ →+ N) ≃+ (M₂ →+ N)` obtained by postcomposition with
+an additive equivalence `e : M₁ ≃+ M₂`."]
+def monoidHomCongrLeft (e : M₁ ≃* M₂) : (M₁ →* N) ≃* (M₂ →* N) where
```

Comment:
My initial suggestion was to drop the `monoidHom` in this name, but I guess this is defensible given that `MulEquiv` would work equally well for `MulHom`.

Can I suggest a zulip thread to bikeshed these names? Should we rename `LinearEquiv.congrLeft` to include the word `linearMap`? Is `LinearEquiv.funCongrLeft` currently correctly named? Are you of the opinion that the names in #25142 are wrong?

Reply:
[Zulip](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Naming.20congruence.20of.20maps)